kernel-spark/src/main/java/io/delta/kernel/spark/read/SparkMicroBatchStream.java

```diff
@@ -183,19 +288,30 @@ CloseableIterator<IndexedFile> getFileChanges(
 
   // TODO(#5318): implement lazy loading (one batch at a time).
   private CloseableIterator<IndexedFile> filterDeltaLogs(
-      long startVersion, Option<DeltaSourceOffset> endOffset) {
+      long startVersion, Optional<DeltaSourceOffset> endOffset) {
     List<IndexedFile> allIndexedFiles = new ArrayList<>();
     // StartBoundary (inclusive)
     CommitRangeBuilder builder =
         TableManager.loadCommitRange(tablePath)
             .withStartBoundary(CommitRangeBuilder.CommitBoundary.atVersion(startVersion));
-    if (endOffset.isDefined()) {
+    if (endOffset.isPresent()) {
       // EndBoundary (inclusive)
       builder =
           builder.withEndBoundary(
               CommitRangeBuilder.CommitBoundary.atVersion(endOffset.get().reservoirVersion()));
     }
-    CommitRange commitRange = builder.build(engine);
+    CommitRange commitRange;
+    try {
+      commitRange = builder.build(engine);
+    } catch (io.delta.kernel.exceptions.KernelException e) {
```

Comment:
You could use https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/exceptions/CommitRangeNotFoundException.java now